### PR TITLE
i3wm.i3-auto-layout: init at 0.2.0

### PR DESCRIPTION
**Pull request by Mephistophiles**
###### Motivation for this change
Adding a new layout helper for i3wm. (https://github.com/chmln/i3-auto-layout)

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/i3/auto-layout.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  cargoSha256 = "0y2qjsn7nd29zdzsrgdadl3bg41xh6lm0r2fj4q0cbyglqmk1k4j";
+
+  # Currently no tests are implemented, so we avoid building the package twice
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Automatic, optimal tiling for i3wm";
+    homepage = "https://github.com/chmln/i3-auto-layout";
+    license = licenses.mit;
+    maintainers = [ ];
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/i3/auto-layout.nix:23]: Adding yourself as a maintainer is required.

File: pkgs/applications/window-managers/i3/auto-layout.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ lib, rustPlatform, pkgs, pkg-config }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "i3-auto-layout";
+  version = "0.2.0";
+
+  src = pkgs.fetchFromGitHub {
+    owner = "chmln";
+    repo = pname;
+    rev = "v0.2";
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/i3/auto-layout.nix:10]: This should reuse version.

File: pkgs/applications/window-managers/i3/auto-layout.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{ lib, rustPlatform, pkgs, pkg-config }:
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/i3/auto-layout.nix:1]:
```suggestion
{ lib, rustPlatform, pkg-config }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
You forgot to include the package.

File: pkgs/applications/window-managers/i3/auto-layout.nix (hunk 2 above)
Comment (Mephistophiles) [reply pkgs/applications/window-managers/i3/auto-layout.nix]: Done

**Comment by Mephistophiles** [comment]
> You forgot to include the package.

Oops, fixed. Thanks for the review!

File: pkgs/applications/window-managers/i3/auto-layout.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{ lib, rustPlatform, fetchFromGitHub, pkg-config }:
```
Comment (dotlambda) [inline pkgs/applications/window-managers/i3/auto-layout.nix:1]:
```suggestion
{ lib, rustPlatform, fetchFromGitHub }:
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
Commit message and pull request title have a typo.

**Comment by Mephistophiles** [comment]
> Commit message and pull request title have a typo.

Fixed, thanks!

File: pkgs/applications/window-managers/i3/auto-layout.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "i3-auto-layout";
+  version = "0.2";
+
+  src = fetchFromGitHub {
+    owner = "chmln";
+    repo = pname;
+    rev = "v{version}";
```
Comment (dotlambda) [inline pkgs/applications/window-managers/i3/auto-layout.nix:10]:
```suggestion
    rev = "v${version}";
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
Does it build for you?

**Comment by Mephistophiles** [comment]
> Does it build for you?

Oh, sorry, the result was cached from the previous build. Fixed and tested on the clean build.

**Comment by Mephistophiles** [comment]
Thanks for the review!

**Pull request merged**